+1 We should bake the default node image into the build. I tried
unsuccessfully

On Fri, Oct 25, 2019 at 11:27 AM Steven Dake <notifications@github.com>
wrote:

> *@sdake* commented on this pull request.
> ------------------------------
>
> In tests/e2e/e2e.sh
> <https://github.com/istio/operator/pull/456#discussion_r339183837>:
>
> >  done
>
> -kubectl get pods --all-namespaces -o wide
> +
> +#kind cluster setup
> +pushd "${ISTIO_DIR}"
> +# shellcheck disable=SC1091
> +source "./prow/lib.sh"
> +setup_kind_cluster kindest/node:v1.15.3
>
> right, my suggestion was to keep this per repo - and then follow this work
> up with a common-files replication. agree re node, we should bake node into
> the buildtools image imo.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/operator/pull/456?email_source=notifications&email_token=AAEYGXOBHBWCZ7ACQHP7UC3QQM3C5A5CNFSM4JESAGR2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJJDV4I#discussion_r339183837>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXIPQLCYCGGEGALZYJTQQM3C5ANCNFSM4JESAGRQ>
> .
>

--- a/deploy/operator.yaml
+++ b/deploy/operator.yaml
@@ -20,7 +20,7 @@ spec:
           command:
           - istio-operator
           - server
-          imagePullPolicy: Always
+          imagePullPolicy: IfNotPresent
           resources:
             limits:
               cpu: 200m

--- a/tests/e2e/e2e.sh
+++ b/tests/e2e/e2e.sh
@@ -17,15 +17,14 @@
 set -eux
 
 export ARTIFACTS="${ARTIFACTS:-$(mktemp -d)}"
-HUB="istio-testing"
-TAG="istio-testing"
+HUB="gcr.io/istio-testing"
+TAG="latest"
 ROOT=$(cd ../../../; pwd)
 
 
 function setup_docker() {
-  HUB=gcr.io/istio-testing TAG=1.5-dev make controller docker
-  kind --loglevel debug --name istio-testing load docker-image gcr.io/istio-testing/operator:1.5-dev
-  kind --loglevel debug --name istio-testing load docker-image istio-testing/app:istio-testing
+  HUB=istio-testing TAG=1.5-dev make controller docker
+  kind --loglevel debug --name istio-testing load docker-image istio-testing/operator:1.5-dev
 }
 
 
@@ -42,7 +41,7 @@ fi
 
 # Create an operator manifest from the default control plane configuration
 cd "${ROOT}/src/istio.io/operator"
-operator_manifest_files=( "deploy/namespace.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_crd.yaml" "deploy/service_account.yaml" "deploy/clusterrole.yaml" "deploy/clusterrole_binding.yaml" "deploy/service.yaml" "deploy/operator.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_cr.yaml" )
+operator_manifest_files=( "deploy/namespace.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_crd.yaml" "deploy/service_account.yaml" "deploy/clusterrole.yaml" "deploy/clusterrole_binding.yaml" "deploy/service.yaml" "tests/e2e/testdata/operator.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_cr.yaml" )
 
 # Generate the main manifest
 rm -f "${ISTIO_DIR}"/install/kubernetes/istio-operator.yaml
@@ -57,11 +56,10 @@ pushd "${ISTIO_DIR}"
 # shellcheck disable=SC1091
 source "./prow/lib.sh"
 setup_kind_cluster ""
-KUBECONFIG=$(kind get kubeconfig-path --name="istio-testing")
-export KUBECONFIG
-HUB="${HUB}" TAG="${TAG}" make docker.app
 popd
 
+KUBECONFIG=$(kind get kubeconfig-path --name="istio-testing")
+export KUBECONFIG
 setup_docker
 
 pushd "${ISTIO_DIR}" || exit

--- a/tests/e2e/e2e.sh
+++ b/tests/e2e/e2e.sh
@@ -17,12 +17,14 @@
 set -eux
 
 export ARTIFACTS="${ARTIFACTS:-$(mktemp -d)}"
+HUB="istio-testing"
+TAG="istio-testing"
 ROOT=$(cd ../../../; pwd)
 
 
 function setup_docker() {
-  HUB=istio-testing TAG=1.5-dev make controller docker
-  kind --loglevel debug --name istio-testing load docker-image istio-testing/operator:1.5-dev
+  HUB=gcr.io/istio-testing TAG=1.5-dev make controller docker
+  kind --loglevel debug --name istio-testing load docker-image gcr.io/istio-testing/operator:1.5-dev
   kind --loglevel debug --name istio-testing load docker-image istio-testing/app:istio-testing
 }
 
@@ -54,20 +56,15 @@ done
 pushd "${ISTIO_DIR}"
 # shellcheck disable=SC1091
 source "./prow/lib.sh"
-setup_kind_cluster kindest/node:v1.15.3
-popd
-
+setup_kind_cluster ""
 KUBECONFIG=$(kind get kubeconfig-path --name="istio-testing")
 export KUBECONFIG
-
-pushd "${ISTIO_DIR}" || exit
-  HUB=istio-testing TAG=istio-testing make docker.app
+HUB="${HUB}" TAG="${TAG}" make docker.app
 popd
 
 setup_docker
 
-
 pushd "${ISTIO_DIR}" || exit
   make istioctl
-  HUB=istio-testing TAG=istio-testing E2E_ARGS="--use_operator --use_local_cluster=true --test_logs_path=${ARTIFACTS}" make e2e_simple_noauth_run
+  HUB="${HUB}" TAG="${TAG}" E2E_ARGS="--use_operator --use_local_cluster=true --test_logs_path=${ARTIFACTS}" make e2e_simple_noauth_run
 popd

--- a/tests/e2e/testdata/operator.yaml
+++ b/tests/e2e/testdata/operator.yaml
@@ -1,0 +1,44 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: istio-operator
+  name: istio-operator
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: istio-operator
+  template:
+    metadata:
+      labels:
+        name: istio-operator
+    spec:
+      serviceAccountName: istio-operator
+      containers:
+        - name: istio-operator
+          image: istio-testing/operator:1.5-dev
+          command:
+          - istio-operator
+          - server
+          imagePullPolicy: IfNotPresent
+          resources:
+            limits:
+              cpu: 200m
+              memory: 256Mi
+            requests:
+              cpu: 50m
+              memory: 128Mi
+          env:
+            - name: WATCH_NAMESPACE
+              value: "istio-operator"
+            - name: LEADER_ELECTION_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: OPERATOR_NAME
+              value: "istio-operator"
+---

